mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -85,7 +86,11 @@ public void configureInputJobProperties(TableDesc tableDesc, Map<String, String>
     map.put(InputFormatConfig.TABLE_IDENTIFIER, props.getProperty(Catalogs.NAME));
     map.put(InputFormatConfig.TABLE_LOCATION, table.location());
     map.put(InputFormatConfig.TABLE_SCHEMA, schemaJson);
+    if (table instanceof BaseTable) {
```

Comment:
Maybe it's better to check the instance of against `HasTableOperations`, in case some Table class other than BaseTable will decide to implement this interface in the future and expose operations on the table

Reply:
I like using `BaseTable` here. I'm not sure that we're going to keep `HasTableOperations` around because it doesn't provide much value over `BaseTable`. If a table uses `TableOperations`, then it is just a thin wrapper so there isn't much of a need to customize.

And metadata tables technically have a `TableOperations` that could be accessible, but replacing the table with a static one based on the metadata location in the table ops would read the table and not the metadata table. So I think it is better to be more specific here.

Reply:
Ended up testing for Serialization as suggested below